src/async_runtime.rs

```diff
@@ -29,43 +30,86 @@ impl AsyncRuntime {
         Ok(Self { rt })
     }
 
-    pub fn handle(&self) -> Handle {
+    pub(crate) fn handle(&self) -> Handle {
         let handle = self.rt.handle().clone();
         Handle { handle }
     }
 }
 
 #[allow(dead_code)]
 impl Handle {
-    pub fn spawn<F>(&self, future: F) -> JoinHandle<F::Output>
+    pub(crate) fn spawn<F>(&self, future: F) -> JoinHandle<F::Output>
     where
         F: Future + Send + 'static,
         F::Output: Send + 'static,
     {
         self.handle.spawn(future)
     }
 
-    pub fn spawn_repeating_task<Func, F>(&self, interval: Duration, func: Func) -> JoinHandle<()>
+    pub(crate) fn spawn_repeating_task<Func, F>(
+        &self,
+        interval: Duration,
+        func: Func,
+    ) -> RepeatingTaskHandle
     where
         Func: Fn() -> F + Send + Sync + 'static,
         F: Future + Send + 'static,
         F::Output: Send + 'static,
     {
-        self.handle.spawn(async move {
+        let (stop_sender, mut stop_receiver) = mpsc::channel(1);
+        let (status_sender, status_receiver) = mpsc::channel(1);
+        let handle = self.handle.spawn(async move {
             let mut interval = time::interval(interval);
             interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+            interval.tick().await;
+
+            // Moved here so that it's dropped when the task finishes
+            let _status_sender = status_sender;
```

Comment:
What about just adding `drop(status_sender);` at the end of the task?

Reply:
Yes, thanks for the suggestion! Fixed in 058f2ed.